SparkPost/2web2ui
# SE-44: Signals ingest batch status view

### What Changed
 - Added `/signals/ingest-batch-status` under **Signals -> Ingestion Status** to display recent ingest batch status events with a simple search interface; similar to events search.

The main elements are:

 - `BatchStatusPage`
     - renders the `Page`
     - controls search and collection subcomponents
     - maintains search state on the query string
     - connected to redux for access to batch status API endpoint
     - owns pagination logic
 - `BatchStatusSearch`
     - renders search controls
     - manages search control state
 - `signalsBatchStatus` actions and reducer
    - pulls events from `/events/ingest`
    - implements cached cursor-driven pagination
 - `useDatePicker` - provides date picker state and controlling props
 - `useQueryParams` - provides qs-based state management
 - `useWindowEvent` - like `WindowEvent` but hooky
 - `useFieldValueOnReturn` - state and onChange callback for a field value, triggered by user hitting return key

### How To Test
 - `npm test` :)
 - Inject some event batches through `/ingest/events`
 - Visit `/signals/ingest-batch-status` and check out the resulting events

### To Do
